src/WebJobs.Script.WebHost/Management/WebFunctionsManager.cs

```diff
@@ -59,22 +59,18 @@ internal static async Task<IEnumerable<FunctionMetadataResponse>> GetFunctionMet
             return await tasks.WhenAll();
         }
 
-        internal IEnumerable<FunctionMetadata> GetFunctionsMetadata(ScriptJobHostOptions hostOptions, ILogger logger, bool includeProxies = false)
+        internal IEnumerable<FunctionMetadata> GetFunctionsMetadata(bool includeProxies, bool forceRefresh)
         {
-            IEnumerable<FunctionMetadata> functionsMetadata = _functionMetadataProvider.GetFunctionMetadata();
+            IEnumerable<FunctionMetadata> functionsMetadata = _functionMetadataManager.GetFunctionMetadata(forceRefresh, includeBlocked: true);
 
-            if (includeProxies)
+            if (!includeProxies)
             {
-                // get proxies metadata
-                var values = ProxyMetadataManager.ReadProxyMetadata(hostOptions.RootScriptPath, logger);
-                var proxyFunctionsMetadata = values.Item1;
-                if (proxyFunctionsMetadata?.Count > 0)
-                {
-                    functionsMetadata = proxyFunctionsMetadata.Concat(functionsMetadata);
-                }
+                // remove proxies metadata
+                functionsMetadata = Utility.FilterOutProxyMetadata(functionsMetadata);
             }
 
-            return functionsMetadata;
+            // Currently we don't want to return codeless functions in any web based APIs
+            return Utility.FilterOutCodeless(functionsMetadata);
```

Comment:
mostly a nit: This would probably be more natural as a predicate we can pass to filter the collection. We can also combine the predicates here so we don't have to iterate over the collection multiple times

Reply:
With the predicate approach, there's no need for Utility functions. I just call the `.Where()`.